### DISCUSSION TITLE
A place to find which Lambda runtimes are supported?

### DISCUSSION THREAD

**Discussion by bjornhansen** [serverless/serverless#12930] [Q&A]
I'm getting errors about unsupported runtimes, though I know Lambda iteself supports newer runtimes. Is there an up-to-date list of supported runtimes for Serverless anywhere and the status of implementing new ones? I can't seem to find one in the documentation or in Github.

For example, **Python 3.13** is supported by Lambda today but I'm getting the error below:

```
at 'provider.runtime': must be equal to one of the allowed values [dotnet6, go1.x, java17, java11, java8, java8.al2, nodejs14.x, nodejs16.x, nodejs18.x, nodejs20.x, provided, provided.al2, provided.al2023, python3.7, python3.8, python3.9, python3.10, python3.11, ruby2.7, ruby3.2]
```

**Comment by MukundaKatta** [comment]
The supported list is baked into the framework's JSON schema, not docs — runtimes land when the schema gets updated. For Python 3.13 specifically, it's added in Serverless v4.4+; upgrading the framework is usually the fix. Until your version catches up, you can bypass validation with `provider.runtime: provided.al2023` + a custom handler layer, but just bumping the framework is cleaner.
